src/noit_filters.c

```diff
@@ -328,6 +339,25 @@ noit_filter_compile_add(mtev_conf_section_t setinfo) {
       rule->type = NOIT_FILTER_SKIPTO;
       rule->skipto = strdup(buffer+strlen(FILTERSET_SKIPTO_STRING));
     }
+    else if (!strncasecmp(buffer, FILTERSET_ADD_MEASUREMENT_TAG_STRING, strlen(FILTERSET_ADD_MEASUREMENT_TAG_STRING))) {
+      rule->type = NOIT_FILTER_ADD_MEASUREMENT_TAG;
+      rule->measurement_tag.add_measurement_tag_cat = NULL;
+      rule->measurement_tag.add_measurement_tag_val = NULL;
+      const char *mt = buffer+strlen(FILTERSET_ADD_MEASUREMENT_TAG_STRING);
+      const char *colon = strstr(mt, ":");
+      if (colon) {
+        size_t cat_len = colon - mt;
+        if (cat_len) {
+          rule->measurement_tag.add_measurement_tag_cat = (char *)calloc(1, cat_len + 1);
+          memcpy(rule->measurement_tag.add_measurement_tag_cat, mt, cat_len);
```

Comment:
``add_measurement_tag_cat`` is null terminated on line 353 but not here

Reply:
It is null terminated. We use `calloc`, which zero-fills the memory. We allocate `cat_len + 1`, then memcpy `cat_len`. That results in an extra allocated byte at the end that is null.